enterprise_extensions/chromatic/chromatic.py

```
@@ -249,15 +251,24 @@ def dm_exponential_dip(tmin, tmax, idx=2, sign="negative", name="dmexp"):
     :param sign:
         set sign of dip: 'positive', 'negative', or 'vary'
     :param name: Name of signal
+    :param vary: Whether to vary the parameters or use constant values.
 
     :return dmexp:
         chromatic exponential dip waveform.
     """
-    t0_dmexp = parameter.Uniform(tmin, tmax)
-    log10_Amp_dmexp = parameter.Uniform(-10, -2)
-    log10_tau_dmexp = parameter.Uniform(0, 2.5)
-    if sign == "vary":
+    if vary:
+        t0_dmexp = parameter.Uniform(tmin, tmax)
+        log10_Amp_dmexp = parameter.Uniform(-10, -2)
+        log10_tau_dmexp = parameter.Uniform(0, 2.5)
+    else:
+        t0_dmexp = parameter.Constant()
+        log10_Amp_dmexp = parameter.Constant()
+        log10_tau_dmexp = parameter.Constant()
+
+    if sign == "vary" and vary:
         sign_param = parameter.Uniform(-1.0, 1.0)
+    elif sign == "vary" and not vary:
+        sign_param = parameter.Constant()
     elif sign == "positive":
         sign_param = 1.0
     else:
```

While you are modifying this, can you make this: `elif sign == "negative"`.  That way an invalid entry won't default to something that runs.  You could optionally `raise` an `ValueError` in a final `else` for invalid input.

---

done